### PR TITLE
Add support for twitter social cards

### PR DESCRIPTION
**Pull request by ribice**

### CODE REVIEW

File: layouts/partials/head/twitter.html [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+<meta name="twitter:site" content="{{ .Site.Params.Settings.pageTwitter}}">
+{{ else }}
+<meta name="twitter:site" content="{{ .Site.Params.social.twitter}}">
+{{ end }}
+<meta name="twitter:creator" content="{{ .Site.Params.social.twitter}}">
+{{ if .IsPage }}
+<meta name="twitter:description" content="{{ .Summary }}" />
+<meta name="twitter:title" content="{{ .Title }}" />
+<meta name="twitter:card" content="summary_large_image" />
+<meta name="twitter:image" content="{{ .Params.image }}" />
```
Comment (ribice) [inline layouts/partials/head/twitter.html]: Perhaps there is a better way to implements this. Currently on my site I use it like this, but for the sake of convenience there might be an easier way.

File: layouts/partials/head/twitter.html [hunk 2]
```diff
@@ -0,0 +1,15 @@
+{{ if .Site.Params.Settings.pageTwitter}}
+<meta name="twitter:site" content="{{ .Site.Params.Settings.pageTwitter}}">
+{{ else }}
+<meta name="twitter:site" content="{{ .Site.Params.social.twitter}}">
+{{ end }}
+<meta name="twitter:creator" content="{{ .Site.Params.social.twitter}}">
+{{ if .IsPage }}
+<meta name="twitter:description" content="{{ .Summary }}" />
+<meta name="twitter:title" content="{{ .Title }}" />
+<meta name="twitter:card" content="summary_large_image" />
```
Comment (ribice) [inline layouts/partials/head/twitter.html]: Perhaps allow this to be set in config, letting the user choose type of card.

File: layouts/partials/head/twitter.html [hunk 3]
```diff
@@ -0,0 +1,15 @@
+{{ if .Site.Params.Settings.pageTwitter}}
+<meta name="twitter:site" content="{{ .Site.Params.Settings.pageTwitter}}">
+{{ else }}
+<meta name="twitter:site" content="{{ .Site.Params.social.twitter}}">
+{{ end }}
+<meta name="twitter:creator" content="{{ .Site.Params.social.twitter}}">
```
Comment (MunifTanjim) [inline layouts/partials/head/twitter.html]: Move `twitter:creator` tag after the [`/layouts/partials/head/opengraph.html` file's 64th line](https://github.com/MunifTanjim/minimo/blob/cd04fa47c0278dc88890c3b583f20a0afd88a56b/layouts/partials/head/opengraph.html#L64).

Let's use the [author's profile data](https://minimo.netlify.com/docs/authors/) for it.

```golang
    {{- with $author.social.twitter -}}
      <meta name='twitter:creator' content='@{{ . }}'>
    {{- end -}}
```

File: layouts/partials/head/twitter.html (hunk 2 above)
Comment (MunifTanjim) [inline layouts/partials/head/twitter.html]: For `twitter:card` tag, we can use this logic:

```html
<meta name='twitter:card' content='summary{{- if .Params.cover -}}_large_image{{- end -}}'>
```

This way, it'll be one less thing for the users to worry about. What do you think about it?

File: exampleSite/config.toml [hunk 4]
```diff
@@ -40,6 +40,7 @@ listDateFormat = "2006, Jan 02"
 hideEntryNavigation = false
 hideSocialMenu = false
 showReadingTime = true
+pageTwitter = "minimo"
```
Comment (MunifTanjim) [inline exampleSite/config.toml]: Let's move this under the `params.opengraph` section:

```toml
[params.opengraph.twitter]
page = ""
```

File: layouts/partials/head/twitter.html [hunk 5]
```diff
@@ -0,0 +1,15 @@
+{{ if .Site.Params.Settings.pageTwitter}}
+<meta name="twitter:site" content="{{ .Site.Params.Settings.pageTwitter}}">
+{{ else }}
+<meta name="twitter:site" content="{{ .Site.Params.social.twitter}}">
```
Comment (MunifTanjim) [inline layouts/partials/head/twitter.html]: For `twitter:site` tag, we should just use `.Site.Params.opengraph.twitter.page`. Is the fallback to `.Site.Params.social.twitter` really necessary?

File: layouts/partials/head/twitter.html [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+<meta name="twitter:creator" content="{{ .Site.Params.social.twitter}}">
+{{ if .IsPage }}
+<meta name="twitter:description" content="{{ .Summary }}" />
+<meta name="twitter:title" content="{{ .Title }}" />
+<meta name="twitter:card" content="summary_large_image" />
+<meta name="twitter:image" content="{{ .Params.image }}" />
+{{ else }}
+<meta name="twitter:title" content="{{ .Site.Title }}" />
+<meta name="twitter:description" content="{{ .Site.Params.info.description }}" />
+{{ end }}
```
[the remark this answers is not in the document]
Comment (MunifTanjim) [reply layouts/partials/head/twitter.html]: And move the whole thing to the end of the **/layouts/partials/head/opengraph.html** file.

File: layouts/partials/head/twitter.html (hunk 5 above)
Comment (ribice) [reply layouts/partials/head/twitter.html]: I think so. If the user only has his personal account set, without the page/website account.

Comment (MunifTanjim) [reply layouts/partials/head/twitter.html]: Still, in that case, the user can set that personal account's username in `.Site.Params.opengraph.twitter.page`.

And as `twitter:site` is a required tag for Twitter Cards, we can check if `.Site.Params.opengraph.twitter.page` is set, and then conditionally render all the twitter tags.

File: layouts/partials/head/twitter.html (hunk 2 above)
Comment (MunifTanjim) [reply layouts/partials/head/twitter.html]: Check this: http://minimo.netlify.com/docs/cover-image/

Comment (ribice) [reply layouts/partials/head/twitter.html]: Ah ok. Thought it was an inbuilt variable.

**Pull request merged**